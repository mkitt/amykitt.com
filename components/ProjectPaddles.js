// @flow
import React from 'react'
import css, { hover } from '../styles/css'
import Button from './Button'
import View from './View'

type Props = {
  onNextDetail: Function,
  onPrevDetail: Function,
}

const style = css({
  position: 'relative',
  textAlign: 'right',
  marginTop: 40,
})

const textStyle = css({
  fontSize: 14,
  color: '#858585',
})

const buttonStyle = css(
  {
    fontSize: 14,
    color: '#858585',
    textTransform: 'uppercase',
    transition: 'color 0.2s',
  },
  hover({ color: '#000' }),
)

export default (props: Props) => (
  <View className={style}>
    <Button className={buttonStyle} onClick={props.onPrevDetail}>
      Prev
    </Button>
    <span className={textStyle}> / </span>
    <Button className={buttonStyle} onClick={props.onNextDetail}>
      Next
    </Button>
  </View>
)
