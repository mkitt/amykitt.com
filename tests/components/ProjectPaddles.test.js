import React from 'react'
import renderer from 'react-test-renderer'
import ProjectPaddles from '../../components/ProjectPaddles'

const noop = () => {}

describe('ProjectPaddles', () => {
  let component

  beforeEach(() => {
    component = (
      <ProjectPaddles
        onNextDetail={noop}
        onPrevDetail={noop}
      />
    )
  })

  it('shows the correct snapshot for the ProjectPaddles component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
