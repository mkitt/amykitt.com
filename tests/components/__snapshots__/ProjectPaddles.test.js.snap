// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPaddles shows the correct snapshot for the ProjectPaddles component tree 1`] = `
<div
  className={
    Object {
      "data-css-1da2mo1": "",
    }
  }
>
  <button
    className="css-1fqszn3 css-12ml4zg"
    onClick={[Function]}
    type="button"
  >
    Prev
  </button>
  <span
    className={
      Object {
        "data-css-1qvhb5d": "",
      }
    }
  >
     / 
  </span>
  <button
    className="css-1fqszn3 css-12ml4zg"
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;
